tests/test_test_completeness.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+import logging
+from tests.common.plugins.test_completeness import CompletenessLevel
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+    pytest.mark.supported_completeness_level(CompletenessLevel.Debug, CompletenessLevel.Basic, \
+        CompletenessLevel.Confident, CompletenessLevel.Thorough)
+]
+
+logger = logging.getLogger(__name__)
+
+
+def test_test_completeness_default(request):
+    defined_levels = [mark.args for mark in request.node.iter_markers(name="completeness_level")]
+    logger.info("Completeness level set: {}".format(str(defined_levels)))
+
+@pytest.mark.supported_completeness_level(CompletenessLevel.Debug)
```

Comment:
I don't think this should be a marker.

By making it a maker, when a completeness level is specified. Test would be selected or deselected. This is not what we want. We want to have this as an argument so that the test could change behavior dynamically.

Reply:
As per the discussion - markers seems to be an acceptable method to implement test-completeness.
Marker will not select or deselect the testcase necessarily - selecting/deselecting depends on whether `pytest.skip` is used.

In the proposed approach, markers enables the testcases to run at different completeness-levels. Global logic to normalize the level if the specified and defined levels do not match will be handled at the test setup level. This prevents duplication of normalization logic in every testcase. Also, after the completeness level is normalized, the testcase is free to decide what to do/execute at different levels of completeness.